### DISCUSSION TITLE
WLAN - RS485 Gateway and RS485 to Actor solution (MQTT over RS485)

### DISCUSSION THREAD

**Discussion by Jaaxy** [arendst/Tasmota#14437] [Q&A] [answered]
Hi, i´m thinking about a Project using 2 ESP to get connection to a place where no WLAN available.

My ESP´s shall working like this:
![image](https://user-images.githubusercontent.com/24737135/149545142-3a7ffedf-6911-4839-890c-60d914bf48d0.png)

Is this possible with ESP-Tasmota? Or is someone working on a solution like this?

In principle, it should work like the Sonoff Gateway, only that instead of 433MHz, MQTT over RS485 should be used here

**Comment by barbudor** [comment, marked as answer]
What is the distance between the 2 devices ?
There a Tasmota features called Range Extender which would create an WIfi AccessPoint on a Tasmota. So if you put the range extender on the edge of your normal Wifi, may be this could extend the wifi up to the other ESP ?

Otherwise, you can use the TasmotaClient feature which is intended to use a serial link between an Arduino platform and a Tasmota.
As of today it is not meant to connect 2 Tasmota togethers but you can program an ESP8266/32 with the Arduino framework and the TasmotaClient library to communication over the RS485 link (the right ESP would not run Tasmota in that case)

Latest option which would be closer to your case is still to use Tasmota on ESP32 on both side with Berry language. Your right ESP could subscribe to topic on the right network (needs it's own MQTT broker) and formward those message through serial to the left ESP.
On the left ESP you receive messages with Berry laguage and you publish to the left network.
(and vice versa if you need bidirectionnal)

**Comment by Jaaxy** [comment]
Too bad, I thought that Tasmota only works from the end device via MQTT over Wifi. But I think it would be a nice feature. I check out the ESP/Arduino Serial Connetion and RS485 as Transmitter.  I have different buildings available with different communications cable connections. I could test that between 30 and 80m.
if it don´t works,  I'll try the whole thing with Lora. Thanks for the answer
